test/integration/042_sources_test/test_sources.py

```diff
@@ -464,6 +469,18 @@ def test_compile(self):
             compiled_sql='select 2 as id'
         )
 
+        macro_override_with_if_statement = self.query(
+            'compile',
+            '{% if True %}select {{ happy_little_macro() }}{% endif %}',
+            name='foo',
+            macros='{% macro override_me() %}2 as id{% endmacro %}'
```

Comment:
are we keeping the `macros` argument to `query`? My understanding was that this argument would go away

Reply:
Yup, I changed `self.query` in this test class to just append the two, instead of making it a parameter: [https://github.com/fishtown-analytics/dbt/blob/0a9ed9977b76f363b6ef988753d6a3b8310608cb/test/integration/042_sources_test/test_sources.py#L325-L333](this) is the new behavior, it just appends macros to any sql it got.